handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -237,36 +237,38 @@ protected void deallocate() {
         keyMaterialManager = context.keyMaterialManager();
         enableOcsp = context.enableOcsp;
 
-        Lock readerLock = context.ctxLock.readLock();
-        readerLock.lock();
-        try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
-        } finally {
-            readerLock.unlock();
-        }
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            Lock readerLock = context.ctxLock.readLock();
+            readerLock.lock();
+            try {
+                ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
Oh yes, sorry.  

I *think* this is okay since the C initialization is preceded by the sync block.  If there were concurrent `ReferenceCountedOpenSslEngine` initializations it seems like it could be a problem, but I don't think that is the case.